src/curvemap.cpp

```diff
@@ -0,0 +1,676 @@
+
+	// no checks required
+
+	return rc;
+}		// PERFORMANCEMAP::pm_CkF
+//-----------------------------------------------------------------------------
+RC PERFORMANCEMAP::pm_GetGridAxis(		// retrieve child grid axis
+	int iGX,		// axis idx (1 based)
+	PMGRIDAXIS*& pGXRet) const
+// returns RCOK iff success (pGXRet set)
+//    else nz RCxxx (msg issued)
+{
+	RC rc = PMGXB.GetIthChild(this, iGX, pGXRet);
+
+	return rc;
+}	// PERFORMANCEMAP::pm_GetGridAxis
+//-----------------------------------------------------------------------------
+RC PERFORMANCEMAP::pm_GetLookupData(
+	int iLU,		// lookup data sought (1 based)
+	PMLOOKUPDATA*& pLURet) const
+{
+	RC rc = PMLUB.GetIthChild(this, iLU, pLURet);
+
+	return rc;
+}	// PERFORMANCEMAP::pm_GetLookupData
+//-----------------------------------------------------------------------------
+RC PERFORMANCEMAP::pm_GXCheckAndMakeVector(
+	int iGX,					// axis idx (1 based)
+	PMGRIDAXIS* &pGX,		// returned: pointer to PMGRIDAXIS if found
+	std::vector< double>& vGX,	// returned: grid values
+	std::pair<int, int> sizeLimits) const	// supported size range
+// returns RCOK iff return values are usable
+//    else non-RCOK, msg(s) issued
+{
+	RC rc = pm_GetGridAxis(iGX, pGX);
+	if (rc)
+		vGX.clear();
+	else
+		rc = pGX->pmx_CheckAndMakeVector(vGX, sizeLimits);
+
+	return rc;
+
+}	// PERFORMANCEMAP::pmx_pm_GXCheckAndMakeVector
+//-----------------------------------------------------------------------------
+RC PERFORMANCEMAP::pm_LUCheckAndMakeVector(
+	int iLU,			// lookup data idx (1 based)
+	PMLOOKUPDATA* &pLU,	// returned: PMLOOKUPDATA found
+	std::vector< double>& vLU,	// returned: values
+	int expectedSize,
+	double vMin /*=-DBL_MAX*/,
+	double vMax /*=DBL_MAX*/) const
+{
+	RC rc = pm_GetLookupData(iLU, pLU);
+	if (rc)
+		vLU.clear();
+	else
+		rc = pLU->pmv_CheckAndMakeVector(vLU, expectedSize, vMin, vMax);
+
+	return rc;
+
+}	// PERFORMANCEMAP::pm_CheckAndMakeVector
+//-----------------------------------------------------------------------------
+RC PERFORMANCEMAP::pm_SetupBtwxt(		// input -> Btwxt conversion
+	record* pParent,		// parent (e.g. RSYS)
+	const char* tag,		// identifying text for this interpolator (for messages)
+	Btwxt::RegularGridInterpolator*& pRgi,		// returned: initialized Btwxt interpolator
+	double& tdbOutRef,				// returned: outdoor dbt for reference values, F
+	double& speedFRef) const		// returned: speed fraction for rated values
+
+// assume pm_type has been checked
+{
+
+	RC rc = RCOK;
+
+	delete pRgi;		// delete prior if any
+	pRgi = nullptr;
+
+	// bool bCooling = pm_type == C_PERFMAPTY_CLGCAPRATCOP;
+
+	// check input data and convert to vector
+	// WHY vector conversion
+	//   * convenient
+	//   * allows modification of values w/o changing input records
+	//     (which may have multiple uses)
+
+	// grid variables
+	std::vector< PMGRIDAXIS*> pGX(2);
+	std::vector< std::vector<double>> vGX(2);	// axis values
+	rc |= pm_GXCheckAndMakeVector( 1, pGX[ 0], vGX[ 0], { 2, 10 });
```

Comment:
Is there a reason to require at least two temperatures? Allowing one, would allow constant performance.